src/orion/benchmark/task/profet/profet_task.py

```diff
@@ -0,0 +1,817 @@
+        self.checkpoint_dir = Path(checkpoint_dir or self.input_dir / "checkpoints")
+        # The config for the training of the meta-model.
+        # NOTE: the train config is used to determine the hash of the task.
+        # TODO: Also save the train_config in the `configuration` of the task.
+        self.train_config = train_config or MetaModelTrainingConfig()
+
+        # The parameters that have an influence over the training of the meta-model are used to
+        # create the filename where the model will be saved.
+        task_hash_params = dict(
+            benchmark=self.benchmark,
+            task_id=self.task_id,
+            seed=self.seed,
+            **asdict(self.train_config),
+        )
+        logger.info(f"Task hash params: {task_hash_params}")
+        task_hash = compute_identity(**task_hash_params)
+
+        filename = f"{task_hash}.pkl"
+
+        self.checkpoint_file = self.checkpoint_dir / filename
+        logger.info(f"Checkpoint file for this task: {self.checkpoint_file}")
+
+        self.device: torch.device
+        if isinstance(device, torch.device):
+            self.device = device
+        else:
+            self.device = torch.device(
+                device or ("cuda" if torch.cuda.is_available() else "cpu")
+            )
+
+        # TODO: Save those in the configuration dict?
+        self._np_rng: Optional[np.random.RandomState] = None
+        self._torch_rng_state: Optional[Tensor] = None
+        with self.seed_randomness():
+            if os.path.exists(self.checkpoint_file):
+                logger.info(
+                    f"Model has already been trained: loading it from file {self.checkpoint_file}."
+                )
+                self.net, self.h = load_task_network(self.checkpoint_file)
+            else:
+                logger.info(
+                    f"Checkpoint file {self.checkpoint_file} doesn't exist: re-training the model."
+                )
+                logger.info(f"Task hash params: {task_hash_params}")
+
+                self.checkpoint_file.parent.mkdir(exist_ok=True, parents=True)
+                # Need to re-train the meta-model and sample this task.
+                self.net, self.h = get_task_network(
+                    self.input_dir,
+                    self.benchmark,
+                    seed=self.seed,
+                    task_id=self.task_id,
+                    config=self.train_config,
+                )
+                save_task_network(
+                    self.checkpoint_file, self.benchmark, self.net, self.h
+                )
+
+        self.net = self.net.to(device=self.device)
+        self.net.eval()
+        self.h_tensor = torch.as_tensor(self.h, dtype=torch.float32, device=self.device)
+
+        self._space: Space = SpaceBuilder().build(self.get_search_space())
+        # IDEA: Could 'hack' the `sample` function of this Space instance so it always uses our
```

Comment:
Yeah eventually it would be great to be able to pass a RNG to the space instead of always passing seeds for each sample. Ideally I think we should not hack around this meanwhile otherwise the code becomes incoherent. Until then we should always assume that the RNG is the global one unless we explicitly pass a seed to sample().

Reply:
Agreed!

Reply:
I create an issue here for that: https://github.com/Epistimio/orion/issues/672. You can remove the comments for this PR. Thanks!

Reply:
Fixed.